java/compat/src/main/java/com/microsoft/bond/compat/CompatDriver.java

```diff
@@ -0,0 +1,65 @@
+package com.microsoft.bond.compat;
+
+import com.microsoft.bond.protocol.CompactBinaryReader;
+import com.microsoft.bond.protocol.CompactBinaryWriter;
+import com.microsoft.bond.protocol.FastBinaryReader;
+import com.microsoft.bond.protocol.FastBinaryWriter;
+import com.microsoft.bond.protocol.ProtocolWriter;
+import com.microsoft.bond.protocol.TaggedProtocolReader;
+import unittest.compat.Compat;
+
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+
+public class CompatDriver {
+    public static void main(final String[] args) throws IOException {
+        if (args.length < 3 || args.length > 4) {
+            System.out.println("Usage:");
+            System.out.println("CompatDriver json|compact|compact2|fast|simple|simple2|schema input_file output_file [json|compact|fast|simple|simple2]");
+            System.exit(255);
+        }
+
+        final String fromProtocol = args[0];
+        final String inputFile = args[1];
+        final String outputFile = args[2];
+        final String toProtocol;
+        if (args.length == 4) {
+            toProtocol = args[3];
+        } else {
+            toProtocol = fromProtocol;
+        }
+
+        final FileInputStream input = new FileInputStream(inputFile);
+        final FileOutputStream output = new FileOutputStream(outputFile);
+        final Compat compat = new Compat();
+
+        TaggedProtocolReader reader = null;
+        ProtocolWriter writer = null;
+
+        if (fromProtocol.equals("fast")) {
+            reader = new FastBinaryReader(input, (short) 1);
+        } else if (fromProtocol.equals("compact")) {
+            reader = new CompactBinaryReader(input, (short) 1);
+        } else if (fromProtocol.equals("compact2")) {
+            reader = new CompactBinaryReader(input, (short) 2);
+        } else {
+            System.err.println("Unsupported input protocol: " + fromProtocol);
+            System.exit(255);
```

Comment:
A number that isn't zero. I have a soft expectation that 255 indicates a usage error and failures following correct invocations start from 1, but don't intend to formalize that until we want a machine to get any info other than success/failure from this program.

Lifted to `static int STATUS_FAILURE`.